src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java

```diff
@@ -149,24 +190,93 @@ public JsonNode postToProcessService(String molfile) throws IOException {
+		//Add the molfile to the hashmap
+		structures.put("mol", molfile);
+		//Post to the service
+		JsonNode jsonNode = postToProcessService(structures);
+
+		return jsonNode;
+	}
+
+	@Override
+	public HashMap<String, String> getPreprocessedStructures(HashMap<String, String> structures)  throws CmpdRegMolFormatException, IOException{
+		// Returns standardized mols
+		String url = getUrlFromPreprocessorSettings("preprocessURL");
+
+		JsonNode jsonNode = getPreprocessorSettings();
+
+		ObjectMapper mapper = new ObjectMapper();
+		ObjectNode requestData = mapper.createObjectNode();
+		JsonNode standardizerActions = jsonNode.get("standardizer_actions");
+		requestData.put("config", standardizerActions);
+		requestData.put("output_format", "MOL");
+
+		// Create a "structures": { "structure-id": "molfile" } object and add it to the request
+		ObjectNode structuresNode = mapper.createObjectNode();
+
+		for (String structureId : structures.keySet()){
+			String structure = structures.get(structureId);
+			structuresNode.put(structureId, structure);
+		}
+		requestData.put("structures", structuresNode);
+
+		// Post to the service
+		HttpURLConnection connection = SimpleUtil.postRequest(url, requestData.toString(), logger);
+		String postResponse = null;
+		if(connection.getResponseCode() != 200) {
+			logger.error("Error posting to preprocessor service: " + connection.getResponseMessage());
+			throw new CmpdRegMolFormatException("Error posting to preprocessor service: " + connection.getResponseMessage());
+		} else {
+			postResponse = SimpleUtil.getStringBody(connection);
+		}
+		logger.debug("Got response: "+ postResponse);
+
+		// Parse the response json to get the standardized mol
+		ObjectMapper responseMapper = new ObjectMapper();
+		JsonNode responseNode = responseMapper.readTree(postResponse);
+		JsonNode structuresResponseNode = responseNode.get("structures");
+
+		HashMap<String, String> standardizedStructures = new HashMap<String, String>();
+		for (String structureId : structures.keySet()){
+			String mol = structuresResponseNode.get(structureId).get("structure").asText();
+			standardizedStructures.put(structureId, mol);
+		}
+		return standardizedStructures;
+	}
+
+	@Override
+	public HashMap<String, BBChemParentStructure> getProcessedStructures(HashMap<String, String> structures, Boolean includeFingerprints) throws CmpdRegMolFormatException {
+
+		// Return map
+		HashMap<String, BBChemParentStructure> processedStructures = new HashMap<String, BBChemParentStructure>();
+		
 		// Post to the service and parse the response
 		try {
 			JsonNode responseNode;
 			try {
-				responseNode = postToProcessService(molfile);
+				responseNode = postToProcessService(structures);
 			} catch (IOException e) {
 				throw new CmpdRegMolFormatException(e);
 			}
-			for (JsonNode responseJsonNode : responseNode)  {
 
+			Object[] structuresArray = structures.keySet().toArray();
+			// Loop through length of response node
```

Comment:
Not actually looping through the structures array. Its a little awkward because the service doesn't take a key value pair of inputs and produce a key value pair of outputs.  Instead they take and array and guarantee the order of the output matches the order of the input (I asked).  So I turned the input into an array and loop through the outputs and assume the order.  I will add a few code comments around this.